From 832010dc4c679a35103d2add5c7e6a28bf5f8b07 Mon Sep 17 00:00:00 2001
From: Romain Perier <mrpouet@gentoo.org>
Date: Thu, 3 Sep 2009 12:35:15 +0200
Subject: [PATCH] Add configure option to use lua system library instead of embedded version

Fixed version: Replaced $enableval by $withval in AC_ARG_WITH(system-lua...),
otherwises it was impossible to exec the configure script correctly with the following options:
--enable-animation --disable-lua --without-system-lua, because LUA was checked anyway.

---
 configure.ac            |   10 ++++
 engines/lua/Makefile.am |  118 +++++++++++++++++++++++++----------------------
 2 files changed, 73 insertions(+), 55 deletions(-)

diff --git a/configure.ac b/configure.ac
index df673ed..6273675 100644
--- a/configure.ac
+++ b/configure.ac
@@ -45,6 +45,9 @@ AC_ARG_ENABLE(paranoia, 	[  --enable-paranoia       use wall, werror, ansi, peda
 AC_ARG_ENABLE(deprecated,	[  --disable-deprecated    disable deprecated functions in gtk et al],	[deprecated=$enableval],		[deprecated="no"])
 
 AC_ARG_ENABLE(schemas,		[  --disable-schema        disable engine schemas],	[schemas=$enableval],	[schemas="yes"])
+AC_ARG_WITH(system-lua,	[  --with-system-lua        link with system Lua library], [system_lua=$withval], [system_lua="no"])
+
+AM_CONDITIONAL([SYSTEM_LUA], [test x$system_lua = xtrue])
 
 BUILD_ENGINES=""
 BUILD_THEMES=""
@@ -126,6 +129,13 @@ AC_SUBST(GTK_LIBS)
 GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
 AC_SUBST(GTK_VERSION)
 
+if test $system_lua = "yes"; then
+	PKG_CHECK_MODULES(LUA, lua,,
+                  AC_MSG_ERROR([--with-system-lua specified but no system liblua found]))
+	AC_SUBST(LUA_CFLAGS)
+	AC_SUBST(LUA_LIBS)
+fi
+
 AC_SUBST(BUILD_ENGINES)
 AC_SUBST(BUILD_THEMES)
 AC_SUBST(BUILD_SCHEMAS)
diff --git a/engines/lua/Makefile.am b/engines/lua/Makefile.am
index 736154d..6243e5a 100644
--- a/engines/lua/Makefile.am
+++ b/engines/lua/Makefile.am
@@ -21,10 +21,69 @@
 
 NULL =
 
+if SYSTEM_LUA
+	LIBLUA_SOURCES =
+	LIBLUA_CFLAGS = $(LUA_CFLAGS)
+else
+	LIBLUA_CFLAGS = -I$(top_srcdir)/engines/lua/src/liblua
+	LIBLUA_SOURCES = \
+		./src/liblua/lapi.h \
+		./src/liblua/lauxlib.h \
+		./src/liblua/lcode.h \
+		./src/liblua/ldebug.h \
+		./src/liblua/ldo.h \
+		./src/liblua/lfunc.h \
+		./src/liblua/lgc.h \
+		./src/liblua/llex.h \
+		./src/liblua/llimits.h \
+		./src/liblua/lmem.h \
+		./src/liblua/lobject.h \
+		./src/liblua/lopcodes.h \
+		./src/liblua/lparser.h \
+		./src/liblua/lstate.h \
+		./src/liblua/lstring.h \
+		./src/liblua/ltable.h \
+		./src/liblua/ltm.h \
+		./src/liblua/lua.h \
+		./src/liblua/luaconf.h \
+		./src/liblua/lualib.h \
+		./src/liblua/lundump.h \
+		./src/liblua/lvm.h \
+		./src/liblua/lzio.h \
+		./src/liblua/lapi.c \
+		./src/liblua/lcode.c \
+		./src/liblua/ldebug.c \
+		./src/liblua/ldo.c \
+		./src/liblua/ldump.c \
+		./src/liblua/lfunc.c \
+		./src/liblua/lgc.c \
+		./src/liblua/liolib.c \
+		./src/liblua/llex.c \
+		./src/liblua/lmem.c \
+		./src/liblua/lobject.c \
+		./src/liblua/lopcodes.c \
+		./src/liblua/loslib.c \
+		./src/liblua/lparser.c \
+		./src/liblua/lstate.c \
+		./src/liblua/lstring.c \
+		./src/liblua/lstrlib.c \
+		./src/liblua/ltable.c \
+		./src/liblua/ltablib.c \
+		./src/liblua/ltm.c \
+		./src/liblua/lundump.c \
+		./src/liblua/lvm.c \
+		./src/liblua/lzio.c \
+		./src/liblua/lauxlib.c \
+		./src/liblua/lbaselib.c \
+		./src/liblua/ldblib.c \
+		./src/liblua/lmathlib.c \
+		./src/liblua/loadlib.c \
+		./src/liblua/linit.c
+endif
+
 INCLUDES = \
 	-I$(top_srcdir)/engines/support \
-	-I$(top_srcdir)/engines/lua/src/liblua \
-	$(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
+	$(LIBLUA_CFLAGS) $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
 	$(NULL)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
@@ -44,62 +103,11 @@ libluaengine_la_SOURCES = \
 	./src/main.c \
 	./src/misc_utils.c \
 	./src/misc_utils.h \
-	./src/liblua/lapi.h \
-	./src/liblua/lauxlib.h \
-	./src/liblua/lcode.h \
-	./src/liblua/ldebug.h \
-	./src/liblua/ldo.h \
-	./src/liblua/lfunc.h \
-	./src/liblua/lgc.h \
-	./src/liblua/llex.h \
-	./src/liblua/llimits.h \
-	./src/liblua/lmem.h \
-	./src/liblua/lobject.h \
-	./src/liblua/lopcodes.h \
-	./src/liblua/lparser.h \
-	./src/liblua/lstate.h \
-	./src/liblua/lstring.h \
-	./src/liblua/ltable.h \
-	./src/liblua/ltm.h \
-	./src/liblua/lua.h \
-	./src/liblua/luaconf.h \
-	./src/liblua/lualib.h \
-	./src/liblua/lundump.h \
-	./src/liblua/lvm.h \
-	./src/liblua/lzio.h \
-	./src/liblua/lapi.c \
-	./src/liblua/lcode.c \
-	./src/liblua/ldebug.c \
-	./src/liblua/ldo.c \
-	./src/liblua/ldump.c \
-	./src/liblua/lfunc.c \
-	./src/liblua/lgc.c \
-	./src/liblua/liolib.c \
-	./src/liblua/llex.c \
-	./src/liblua/lmem.c \
-	./src/liblua/lobject.c \
-	./src/liblua/lopcodes.c \
-	./src/liblua/loslib.c \
-	./src/liblua/lparser.c \
-	./src/liblua/lstate.c \
-	./src/liblua/lstring.c \
-	./src/liblua/lstrlib.c \
-	./src/liblua/ltable.c \
-	./src/liblua/ltablib.c \
-	./src/liblua/ltm.c \
-	./src/liblua/lundump.c \
-	./src/liblua/lvm.c \
-	./src/liblua/lzio.c \
-	./src/liblua/lauxlib.c \
-	./src/liblua/lbaselib.c \
-	./src/liblua/ldblib.c \
-	./src/liblua/lmathlib.c \
-	./src/liblua/loadlib.c \
-	./src/liblua/linit.c \
+	$(LIBLUA_SOURCES) \
 	$(NULL)
 
 libluaengine_la_LDFLAGS = -module -avoid-version -no-undefined -export-symbols $(top_srcdir)/engines/engine.symbols
-libluaengine_la_LIBADD = $(top_builddir)/engines/support/libsupport.la $(GTK_LIBS)
+libluaengine_la_LIBADD = $(top_builddir)/engines/support/libsupport.la $(GTK_LIBS) $(LUA_LIBS)
 
 -include $(top_srcdir)/git.mk
 
-- 
1.6.4.2

